### PR TITLE
Issue 1184: Drop the deprecated validateAndStore method from the ArtifactManagementService

### LINKED ISSUE #1184: Drop the deprecated validateAndStore method from the ArtifactManagementService
# Task Description

We need to drop the `@Deprecated` method `validateAndStore(String storageId, String repositoryId, String path, InputStream is)` from the `ArtifactManagementService` and replace all usages of it with `validateAndStore(RepositoryPath repositoryPath, InputStream is)`:
```
    @Deprecated
    @Transactional
    public long validateAndStore(String storageId,
                                 String repositoryId,
                                 String path,
                                 InputStream is)
            throws IOException,
                   ProviderImplementationException,
                   NoSuchAlgorithmException,
                   ArtifactCoordinatesValidationException
    {
        RepositoryPath repositoryPath = repositoryPathResolver.resolve(storageId,
                                                                       repositoryId,
                                                                       path);

        return validateAndStore(repositoryPath, is);
    } 
```

Also, the replacement method no longer throws a `NoSuchAlgorithmException`, so that should be removed from the list of thrown exceptions.

# Acceptance Test

* [ ] Building the code with `mvn clean install -Dintegration.tests` still works.
* [ ] Running `mvn spring-boot:run` in the `strongbox-web-core` still starts up the application correctly.
* [ ] Building the code and running the `strongbox-distribution` from a `zip` or `tar.gz` still works.
* [ ] The tests in the [`strongbox-web-integration-tests`](https://github.com/strongbox/strongbox-web-integration-tests/) still run properly.

# Help

* [Our chat](https://chat.carlspring.org/)
* Points of contact:
  * @carlspring
  * @sbespalov
  * @fuss86

### PR DESCRIPTION
**Pull request by akhil10x5**
fixes #1184

### CODE REVIEW

**Comment by akhil10x5** [comment]
Jenkins job failed with following error:
[2019-04-07T12:36:27.196Z] [ERROR] Failed to execute goal on project strongbox-authentication-api: Could not resolve dependencies for project org.carlspring.strongbox:strongbox-authentication-api:jar:1.0-SNAPSHOT: Could not transfer artifact org.apache.logging.log4j:log4j-api:jar:2.11.1 from/to carlspring (http://repo.carlspring.org/content/groups/carlspring/): /cache/jenkins-strongbox-strongbox-PR-1186/.m2/repository/org/apache/logging/log4j/log4j-api/2.11.1/log4j-api-2.11.1.jar.part (No such file or directory) -> [Help 1]

I guess this is not due my code changes :)

File: strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout/strongbox-storage-maven-layout-provider/src/test/java/org/carlspring/strongbox/providers/search/MavenIndexerSearchProviderTest.java [hunk 1]
```diff
@@ -148,10 +153,12 @@ public void shouldBeCapableToSearchByFullSha1Checksum(TestInfo testInfo)
     public void shouldBeCapableToSearchByClassNameByDefault(TestInfo testInfo)
             throws Exception
     {
-        artifactManagementService.validateAndStore(STORAGE0,
-                                                   getRepositoryName(REPOSITORY_RELEASES_WITH_CLASSNAMES_INDEXED,
-                                                                     testInfo),
-                                                   "org/carlspring/properties-injector/1.7/properties-injector-1.7.jar",
+    	
+    	RepositoryPath repositoryPath = repositoryPathResolver.resolve(STORAGE0,
```
Comment (carlspring) [inline strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout/strongbox-storage-maven-layout-provider/src/test/java/org/carlspring/strongbox/providers/search/MavenIndexerSearchProviderTest.java]: Cosmetic: Could you please follow the project's coding convention and align these method parameters properly (here and across thePR)?

**Comment by carlspring** [comment]
@akhil10x5,

Thank you for your contribution!

Would you mind signing the ICLA, as described in the [Contributing](https://github.com/strongbox/strongbox/blob/master/CONTRIBUTING.md) page?

Also, the username under which you're committing and pushing to Github seems to be mis-configured. You will have to look into how to configure your git settings properly, amend the author of the commits and then force push back to the branch. (When your user is properly set up, you see your avatar when against your commits in the commit history, otherwise it's a default grey avatar).

File: strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout/strongbox-storage-maven-layout-provider/src/test/java/org/carlspring/strongbox/providers/search/MavenIndexerSearchProviderTest.java (hunk 1 above)
Comment (akhil10x5) [reply strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout/strongbox-storage-maven-layout-provider/src/test/java/org/carlspring/strongbox/providers/search/MavenIndexerSearchProviderTest.java]: Done!

**Comment by carlspring** [comment]
@akhil10x5 ,

Thanks for fixing the build! Could you now please sort out your git/github username/avatar and amend the author for the commits in this pull? Thanks! :)

**Comment by carlspring** [comment]
Thank you for your help and congratulations on your first pull request!

**Pull request merged**